apps/worker/src/app/workflow/usecases/send-message/send-message.usecase.ts

```diff
@@ -223,6 +235,36 @@ export class SendMessage {
     return result;
   }
 
+  @Instrument()
+  private async buildCompileContext(command: SendMessageCommand): Promise<IFilterVariables> {
+    const [subscriber, actor, tenant] = await Promise.all([
+      this.getSubscriberBySubscriberId({
+        subscriberId: command.subscriberId,
+        _environmentId: command.environmentId,
+      }),
+      command.job.actorId &&
+        this.getSubscriberBySubscriberId({
+          subscriberId: command.job.actorId,
+          _environmentId: command.environmentId,
+        }),
+      this.handleTenantExecution(command.job),
+    ]);
+
+    if (!subscriber) throw new PlatformException('Subscriber not found');
+
+    return {
+      subscriber,
+      payload: command.payload,
+      step: {
+        digest: !!command.events?.length,
+        events: command.events,
+        total_count: command.events?.length,
+      },
+      ...(tenant && { tenant }),
+      ...(actor && { actor }),
+    };
```

Comment:
This is used in all send message usecases and now in the filters as well. Extracted this object so we fetch and build it once.